packages/@magic-sdk/types/src/modules/auth-types.ts

```diff
@@ -278,12 +280,14 @@ export type LoginWithEmailOTPEventHandlers = {
   [LoginWithEmailOTPEventOnReceived.InvalidMfaOtp]: () => void;
   [LoginWithEmailOTPEventOnReceived.ExpiredEmailOtp]: () => void;
   [LoginWithEmailOTPEventOnReceived.MfaSentHandle]: () => void;
+  [LoginWithEmailOTPEventOnReceived.InvalidRecoveryCode]: () => void;
   [AuthEventOnReceived.IDTokenCreated]: (idToken: string) => void;
   [WalletEventOnReceived.WalletInfoFetched]: () => void;
 
   // Event sent
   [LoginWithEmailOTPEventEmit.VerifyEmailOtp]: (otp: string) => void;
   [LoginWithEmailOTPEventEmit.VerifyMFACode]: (mfa: string) => void;
+  [LoginWithEmailOTPEventEmit.LostDevice]: (recoveryCode: string) => void;
```

Comment:
Instead of having just one LostDevice event, I suggest we split them into two.
LostDevice and `submitRecoveryCode`

By doing this, it shall make Mandrake implementation easier